utils/lru.go

```diff
@@ -155,6 +156,20 @@ func (c *Cache) Remove(key interface{}) {
 	}
 }
 
+// RemoveStringPrefix removes string keys starting with prefix from the cache.
+func (c *Cache) RemoveStringPrefix(prefix string) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	for _, key := range c.keys() {
```

Comment:
This became linear time in the size of the cache... so not sure this is a great solution. Do we have to support arbitrary limits? Maybe we should just match the client's common case and do just 30.

Reply:
At first my thinking was sure it's linear time but how many cache entires can there be? Then I did the napkin math and the worst case is 1000 entries per channel and if you have 2000 channels (not unheard of on large instances) then entries are getting pretty high into the millions.

So yeah I guess it's better to just support a small whitelist (30 and 60)